pkg/api/api.go

```diff
@@ -112,23 +112,28 @@ type apiHandler func(r *http.Request) (interface{}, error)
 
 func wrap(h apiHandler) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
+		defer r.Body.Close()
 		ctx := r.Context()
 		r = r.WithContext(log.WithRequestID(ctx, middleware.GetReqID(ctx)))
 		defer func() {
-			_ = log.RequestIDLogger(r).Sync()
+			_ = log.RequestIDLogger(ctx).Sync()
 		}()
 
 		respObj, err := h(r)
 		if err != nil {
 			writeError(w, err)
+			return
```

Comment:
Returning here will cause a 500 error reported to the client with nothing in the HTTP response body. I'd think we would want to return some sort of an error message back to the client (which the prior code supports, but is unused). Thoughts?